src/plugins/chart_expressions/expression_partition_vis/public/utils/get_columns.ts

```diff
@@ -27,20 +27,29 @@ export const getColumns = (
 } => {
   const { metric, buckets } = visParams.dimensions;
   if (buckets && buckets.length > 0) {
-    const bucketColumns: Array<Partial<BucketColumns>> = buckets.map(({ accessor, format }) => ({
-      ...getColumnByAccessor(accessor, visData.columns),
-      format,
-    }));
+    const bucketColumns: Array<Partial<BucketColumns>> = buckets.map((bucket) => {
+      const column = getColumnByAccessor(bucket, visData.columns);
+      return {
+        ...column,
+        format: typeof bucket === 'string' ? column?.meta.params : bucket.format,
+      };
+    });
 
     const lastBucketId = bucketColumns[bucketColumns.length - 1].id;
     const matchingIndex = visData.columns.findIndex((col) => col.id === lastBucketId);
 
     return {
       bucketColumns,
-      metricColumn: getMetricColumn(metric?.accessor ?? matchingIndex + 1, visData),
+      metricColumn: getMetricColumn(
+        metric ?? ({ accessor: matchingIndex + 1 } as ExpressionValueVisDimension),
```

Comment:
Please, replace `as ExpressionValueVisDimension` with the value, matching this type.